# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/systemservices/CamerasChild.cpp
# Commit: fef894fcc429
# Full Hash: fef894fcc4294027fdfb62af3997b87411bef554
# Author: Jan Grulich <jgrulich@redhat.com>
# Date: 2023-11-30 17:23:44
# Regressor Bug: 1844020
# File Overlap Count: 1
# Description:
#   Bug 1844020 - Add a camera device placeholder in case PipeWire is not initialized yet r=pehrsons
#   
#   Implement DeviceInfoPipeWire with option to add a device placeholder
#   when there is a camera device, but we don't have access to PipeWire yet.
#   We ask the Camera portal whether there is a camera device present.
# ==============================================================================

diff -r ed31b2acb5fb -r fef894fcc429 dom/media/systemservices/CamerasChild.cpp
--- a/dom/media/systemservices/CamerasChild.cpp	Thu Nov 30 11:49:35 2023 +0000
+++ b/dom/media/systemservices/CamerasChild.cpp	Thu Nov 30 11:49:36 2023 +0000
@@ -289,10 +289,13 @@
   return IPC_OK();
 }
 
-int CamerasChild::GetCaptureDevice(
-    CaptureEngine aCapEngine, unsigned int list_number, char* device_nameUTF8,
-    const unsigned int device_nameUTF8Length, char* unique_idUTF8,
-    const unsigned int unique_idUTF8Length, bool* scary) {
+int CamerasChild::GetCaptureDevice(CaptureEngine aCapEngine,
+                                   unsigned int list_number,
+                                   char* device_nameUTF8,
+                                   const unsigned int device_nameUTF8Length,
+                                   char* unique_idUTF8,
+                                   const unsigned int unique_idUTF8Length,
+                                   bool* scary, bool* device_is_placeholder) {
   LOG(("%s", __PRETTY_FUNCTION__));
   nsCOMPtr<nsIRunnable> runnable =
       mozilla::NewRunnableMethod<CaptureEngine, unsigned int>(
@@ -306,6 +309,9 @@
     if (scary) {
       *scary = mReplyScary;
     }
+    if (device_is_placeholder) {
+      *device_is_placeholder = mReplyDeviceIsPlaceholder;
+    }
     LOG(("Got %s name %s id", device_nameUTF8, unique_idUTF8));
   }
   return dispatcher.ReturnValue();
@@ -313,7 +319,7 @@
 
 mozilla::ipc::IPCResult CamerasChild::RecvReplyGetCaptureDevice(
     const nsACString& device_name, const nsACString& device_id,
-    const bool& scary) {
+    const bool& scary, const bool& device_is_placeholder) {
   LOG(("%s", __PRETTY_FUNCTION__));
   MonitorAutoLock monitor(mReplyMonitor);
   mReceivedReply = true;
@@ -321,6 +327,7 @@
   mReplyDeviceName = device_name;
   mReplyDeviceID = device_id;
   mReplyScary = scary;
+  mReplyDeviceIsPlaceholder = device_is_placeholder;
   monitor.Notify();
   return IPC_OK();
 }